ocaml/typing/persistent_env.ml

```diff
@@ -83,6 +112,20 @@ type import_info =
   | Missing
   | Found of import
 
+(* Data relating to a global name (possibly with arguments) but not necessarily
+   a value in scope. For example, if we've encountered a parameter module only
+   by seeing it used as the name of an argument in a [Global_module.Name.t], we
+   won't bind it or construct a [pers_struct] for it but it will have a
+   [pers_name]. *)
```

Comment:
I'll update the comment: A better example is when we've encountered a module as the _value_ of an argument in an instance name. We need its import info and its signature but we don't actually need a binding for it. For parameter names, we might consider trying to cache less about them but I don't see much reason to have four caches rather than three (`pn_arg_for` is the only thing that a parameter _can't have_ and it already has `None` as a sensible value there).

Reply:
> A better example is when we've encountered a module as the value of an argument in an instance name. We need its import info and its signature but we don't actually need a binding for it.

Hmm, if in the source code I have `B[A/P]` as instance name, I should probably have `A` in scope? For example, the signature of `B` can mention `P.t`, it would now mentions `A.t`?

Reply:
`A` might be in scope in some sense, but we don't necessarily need to bind it as a runtime parameter unless we actually need to compile a reference to its value. That's really where the dividing line between `pers_name` and `pers_struct` is: `pers_name` is for typing and `pers_struct` is for actually generating lambda code.

Reply:
That makes sense, thanks. But in that case, we probably don't need `pn_sign`? That is, we can delay generating the signature until `pers_struct`.

Reply:
So, offline we decided to keep `Signature_with_global_bindings` doing what it does, which I think means keeping `pn_sign` since as you say it's entangled with bound globals? At any rate there's probably no major savings here, since `pn_sign` gets computed lazily anyway.